paymentsheet/src/main/java/com/stripe/android/customersheet/InternalCustomerSheetResult.kt

```diff
@@ -4,34 +4,43 @@ import android.content.Intent
 import android.os.Bundle
 import android.os.Parcelable
 import androidx.core.os.bundleOf
-import com.stripe.android.paymentsheet.model.PaymentOption
+import com.stripe.android.paymentsheet.model.PaymentOptionFactory
+import com.stripe.android.paymentsheet.model.PaymentSelection
 import com.stripe.android.view.ActivityStarter
 import kotlinx.parcelize.Parcelize
 
 @OptIn(ExperimentalCustomerSheetApi::class)
 internal sealed class InternalCustomerSheetResult : Parcelable {
-    abstract fun toPublicResult(): CustomerSheetResult
+    abstract fun toPublicResult(
+        paymentOptionFactory: PaymentOptionFactory,
+    ): CustomerSheetResult
 
     /**
      * The customer selected a payment method
      */
     @Parcelize
-    class Selected internal constructor(
-        val paymentMethodId: String,
-        val drawableResourceId: Int,
-        val label: String,
+    data class Selected internal constructor(
+        val paymentSelection: PaymentSelection?
     ) : InternalCustomerSheetResult() {
-        @Suppress("DEPRECATION")
-        override fun toPublicResult(): CustomerSheetResult {
-            return CustomerSheetResult.Selected(
-                selection = PaymentOptionSelection(
-                    paymentMethodId = paymentMethodId,
-                    // TODO (jameswoo) Use [PaymentOptionFactory]
-                    paymentOption = PaymentOption(
-                        drawableResourceId = drawableResourceId,
-                        label = label,
+        override fun toPublicResult(
+            paymentOptionFactory: PaymentOptionFactory,
+        ): CustomerSheetResult {
+            val selection = when (paymentSelection) {
```

Comment:
Should we pull this mapping out to a function? It looks copy pasted between the two.